proposals/NNNN-groupshared-arguments.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "NNNN - `groupshared` Arguments"
+params:
+  authors:
+    - llvm-beanz: Chris Bieneman
+  sponsors:
+    - tbd: TBD
+  status: Under Consideration
+---
+
+* Planned Version: 202x
+
+## Introduction
+
+This proposal introduces a new use of the `groupshared` keyword for function
+arguments to allow passing `groupshared` arguments by address rather than by
+value or copy-in/copy-out.
+
+## Motivation
+
+DXC's implementation of HLSL includes a set of Interlocked functions
+implementing atomic operations on `groupshared` memory. These functions are not
+expressible in HLSL and rely on special case implementation in DXC.
+
+## Proposed solution
+
+HLSL 202x will allow the `groupshared` type annotation keyword on function
+parameter declarations. The keyword when applied to a parameter declaration of
+type `T`, alters the qualified type of the parameter to a `groupshared & T`
```

Comment:
nit/question: Should the altered type rather be `groupshared T &`?
Thinking of C++, where it would look like `groupshared T &myvar` with identifier.